homeassistant/components/zwave_js/sensor.py

```diff
@@ -581,26 +530,6 @@ def device_class(self) -> SensorDeviceClass | None:
             return SensorDeviceClass.ENUM
         return None
 
-    @property
-    def options(self) -> list[str] | None:
-        """Return options for enum sensor."""
-        if self.device_class == SensorDeviceClass.ENUM:
-            return list(self.info.primary_value.metadata.states.values())
-        return None
-
-    @property
-    def native_value(self) -> str | None:
-        """Return state of the sensor."""
-        if self.info.primary_value.value is None:
-            return None
-        key = str(self.info.primary_value.value)
-        if (
-            self._primary_value.configuration_value_type == ConfigurationValueType.RANGE
```

Comment:
Why don't we need this check anymore?

Reply:
it's a no-op. If the value is of type `RANGE`,  `allowManualEntry` is false and there are no states. The logic table below shows the intended behavior, and this check is redundant as a result.

| ConfigurationValueType | `allowManualEntry` value | non-empty `states` | `key` in `states`? | `native_value` |
|---|---|---|---|---|
| RANGE | False | False | NEVER | `key` |
| MANUAL_ENTRY | True | False | NEVER | `key` |
| MANUAL_ENTRY | True | True | SOMETIMES | `key` or `value` depending on whether `key` is in `states` |
| ENUM | False | True | ALWAYS | `value` |